client/verta/verta/client.py

```diff
@@ -621,29 +652,54 @@ def set_experiment(self, name=None, desc=None, tags=None, attrs=None, id=None):
 
         name = self._set_from_config_if_none(name, "experiment")
 
+        if id is None and self._ctx.proj is None:
+            self.set_project()
```

Comment:
This is pulled from the old lines 634–635 below, because the static method can't access `self.set_project()`